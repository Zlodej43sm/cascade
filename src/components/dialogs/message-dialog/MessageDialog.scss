@use '../../../styles/sizes';

.container {
  margin-bottom: sizes.$base-3;

  > label:not(:last-of-type) {
    margin-bottom: sizes.$base-2;
  }

  textarea {
    height:  sizes.$textarea-dialog-height;
    resize: none;
  }
}

.footer {
  text-align: right;
}
