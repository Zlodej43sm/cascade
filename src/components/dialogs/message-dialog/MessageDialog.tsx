import React from 'react';

import DialogWrapper, { DialogWrapperProps } from 'lib/components/dialog-wrapper/DialogWrapper';
import InputField from 'lib/components/forms/input-field/InputField';
import TextAreaField from 'lib/components/forms/text-area-field/TextAreaField';
import Button, { ButtonSizes } from 'lib/components/buttons/button/Button';

import styles from './MessageDialog.scss';

enum FieldProps {
  MemberInputName = 'member',
  MemberInputLabel = 'Members',
  MemberInputPlaceholder = 'Please enter member name',
  MessageName = 'message',
  MessageTextareaLabel = 'Message',
  MessageTextareaPlaceholder = 'Can I please get an update today?',
}

export type MessageDialogProps = DialogWrapperProps & {
  onSubmit?: (event: React.MouseEvent<HTMLButtonElement>) => void
};

function MessageDialog({ onSubmit, ...wrapperProps }: MessageDialogProps) {
  return (
    <DialogWrapper title="Message" width={568} {...wrapperProps}>
      <div className={styles.container}>
        <InputField
          name={FieldProps.MemberInputName}
          label={FieldProps.MemberInputLabel}
          placeholder={FieldProps.MemberInputPlaceholder}
        />
        <TextAreaField
          name={FieldProps.MessageName}
          label={FieldProps.MessageTextareaLabel}
          placeholder={FieldProps.MessageTextareaPlaceholder}
        />
      </div>
      <footer className={styles.footer}>
        <Button onClick={onSubmit} size={ButtonSizes.Compact}>Send</Button>
      </footer>
    </DialogWrapper>
  );
}

export default MessageDialog;
