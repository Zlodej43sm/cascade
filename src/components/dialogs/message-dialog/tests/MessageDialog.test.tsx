import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { messageDialogArguments } from '../__mocks__/MessageDialog.mock';
import MessageDialog from '../MessageDialog';

describe('MessageDialog', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<MessageDialog {...messageDialogArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Should call apply handler', () => {
    const onSubmit = jest.fn();
    const { getByTestId } = render(<MessageDialog {...messageDialogArguments} onSubmit={onSubmit} />);
    fireEvent.click(getByTestId('button'));
    expect(onSubmit).toHaveBeenCalled();
  });
});
