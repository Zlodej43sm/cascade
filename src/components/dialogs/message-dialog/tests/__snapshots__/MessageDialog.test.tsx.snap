// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageDialog Should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="dialog-wrapper-container"
    style="width: 568px;"
  >
    <header
      class="header"
    >
      <div
        class="title"
        data-testid="dialog-wrapper-title"
      >
        Message
      </div>
      <div
        class="closeButton"
      >
        <button
          aria-label="Close"
          class="button"
          data-testid="close-button"
          type="button"
        />
      </div>
    </header>
    <div
      class="content"
    >
      <div
        class="container"
      >
        <label
          class="wrapper"
        >
          <span
            class="label"
          >
            Members
          </span>
          <input
            class="input"
            name="member"
            placeholder="Please enter member name"
          />
        </label>
        <label
          class="wrapper"
        >
          <span
            class="label"
          >
            Message
          </span>
          <textarea
            class="input"
            name="message"
            placeholder="Can I please get an update today?"
          />
        </label>
      </div>
      <footer
        class="footer"
      >
        <button
          class="button primary compact"
          data-testid="button"
          type="button"
        >
          Send
        </button>
      </footer>
    </div>
  </section>
</DocumentFragment>
`;
